native_client/deepspeech.h

```diff
@@ -1,5 +1,5 @@
-#ifndef __DEEPSPEECH_H__
-#define __DEEPSPEECH_H__
+#ifndef DEEPSPEECH_H
```

Comment:
Currently all of those cause error messages followed by a segfault (since they're all checked in the Model constructor). Do you mean deal with error conditions in the API?

Reply:
Yes, for example, `aModelPath` is checked in the constructor. If there is a problem, an error message is printed and an invalid `Model` is constructed.

So _programmatically_ there's no way, other than grepping the error logs, for a client of this API to deal with such errors.

For example, say if Deep Speech were being used in a Scout, Mozilla newest hypothetical voice assistant, which has no screen only a voice based UI. Somehow during a model update, say, the path is set incorrectly. How does Scout know this? Grepping the error log? This seems like the wrong thing to force on Scout.

Scout should be able to catch C++ exceptions, likely not the right solution as, for example, RAII might no be consistently used, or check for error codes, much easier and lower risk, or do something else. But forcing a program to grep the error log seems like the wrong thing to do. 

So in summary I am saying we must deal with error conditions in the API or the API can't really be used in production systems that have to handle errors gracefully.